### PR TITLE
Create PR deploys

### PR DESCRIPTION
**Pull request by kim0**
Fix https://github.com/brave/leo/issues/103

### CODE REVIEW

File: .github/workflows/build.yml
Comment (AlanBreck) [inline .github/workflows/build.yml:81]: Since this is public, we can remove the part about credentials.

```suggestion
              body: `👋 Thanks for Submitting! This PR is available for preview at the link below.\n\n✅ Preview: https://leo.bravesoftware.com/pr-${context.issue.number}/${context.payload.after}`
```

**Review by AlanBreck — DISMISSED** [review]

**Comment by AlanBreck** [comment]
Is it possible to have the PR number in the subdomain as we do with Bat.org and Brave.com?

**Comment by petemill** [comment]
I'm on VPN and being prompted for basic auth credentials, but the ones I'm using from `"Brave.com Staging/Previews"` aren't being accepted.

**Comment by kim0** [comment]
@petemill I'll remove the authentication function since @AlanBreck explained this should be public

**Comment by AlanBreck** [comment]
Looks really good, but getting a 404 on a JS file.
Screenshot 2022-12-07 at 3 13 06 PM

**Comment by fallaciousreasoning** [comment]
Apart from the 404 this is great - thanks heaps @kim!

**Comment by kim0** [comment]
@fallaciousreasoning @AlanBreck the 404 should be fixed now. Let me know if anything else is still needed

**Comment by fallaciousreasoning** [comment]
I still get a 404 on 121.pr.leo.bravesoftware.com (leo.bravesoftware.com & https://121.pr.leo.bravesoftware.com/commit-fa7a71d67cb6cd12a24f9c599c9dfdcd9673bd9f/ work fine though)

**Comment by fallaciousreasoning** [comment]
Works in a private tab, so I think you're right :)

**Comment by fallaciousreasoning** [comment]
Okay, this LGTM, but I'll let @petemill or @AlanBreck add the +1. Cheers for setting this up @kim0!

**Comment by AlanBreck** [comment]
Looks good to me as well! Biggest question would be the CSP.

@petemill, do we have a CSP on the Brave core storybook?

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml]: Good catch, I've removed this debugging line

**Comment by fallaciousreasoning** [comment]
@AlanBreck maybe we should merge this as is and look into the CSP later?

**Comment by kim0** [comment]
@brave/sec-team please advise if CSP should be regarded as a blocker for merging this PR. If not, it can be handled in its own issue I suppose. Thanks!

cc: @AlanBreck

**Comment by kdenhartog** [comment]
I don't think a CSP needs to be a blocker for this particular PR, but it definitely seems like something we want to do for this ASAP since the Storybook page is public.

**Comment by kdenhartog** [comment]
@brave/devops-team can one of you double check that this requires proper permissions on GH so that people not apart of the org require approval to run the CI pipeline? I want to make sure that we're preventing an issue where a contributor is adding a malicious story book component which is getting hosted under the `bravesoftware.com` domain

**Comment by kim0** [comment]
@kdenhartog thanks for the comments, here are my replies
* Only collaborators with write access to this repo can open PRs that will run the workflow
* PRs coming from forks (external collaborators) approval mode is set to `Require approval for all outside collaborators` as can be seen in the screenshot below and in [settings/actions](https://github.com/brave/leo/settings/actions) page.
* In case approval for even those who have write access to the repo is needed, the deployment envrionment can be configured to require reviewers which can be set [here](https://github.com/brave/leo/settings/environments/749217889/edit).

Screenshot 2022-12-13 at 11 12 00 AM

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:67]: We use cloud posse modules to create s3-cdn services. The devops PR is https://github.com/brave/devops/pull/8771
You can view the bucket policy by logging into the `web-stg` account and clicking [here](https://s3.console.aws.amazon.com/s3/buckets/brave-staging-leo-origin?region=us-west-2&tab=permissions)
But probably the interesting piece is the github actions iam policy defined [here](https://s3.console.aws.amazon.com/s3/buckets/brave-staging-leo-origin?region=us-west-2&tab=permissions)

Comment (kim0) [reply .github/workflows/build.yml:67]: So basically, if a collaborator with write access opens a PR, this runs the workflow, which does OIDC authentication and gets AWS credentials with the above mentioned IAM policy which allows the workflow write access to the s3 bucket

cc: @mihaiplesa

**Review by fallaciousreasoning — APPROVED** [review]
This LGTM still. Thanks @kdenhartog & @kim0

**Review by AlanBreck — COMMENTED** [review]
Looks good to me!

Will await @kdenhartog sign off from answers above, and then @kim0 can merge at will!

**Review by AlanBreck — APPROVED** [review]

**Review by kdenhartog — APPROVED** [review]
LGTM, the IAM policy looks good and repo configurations looks good, so all my concerns were already addressed.

Also, I just realized I pinged devops when @kim0 is on the team (and the author of the PR), sorry about the mistaking you for being on a different team.

File: .github/workflows/build.yml
Comment (mihaiplesa) [inline .github/workflows/build.yml:73]: Newline before and to add name.

File: .github/workflows/build.yml
Comment (mihaiplesa) [inline .github/workflows/build.yml:55]: Can go 1 up to be sorted alphabetically.

File: .github/workflows/build.yml
Comment (mihaiplesa) [inline .github/workflows/build.yml:44]: `AWS`

**Review by mihaiplesa — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml
Comment (wknapik) [inline .github/workflows/build.yml:57]: I'd suggest
```bash
shopt -s inherit_errexit
set -eEo pipefail
```
instead

File: .github/workflows/build.yml
Comment (wknapik) [inline .github/workflows/build.yml:60]: I think `|| true` is more readable than `|| :`, especially to those less familiar with shell scripts

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:84]: Done

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:73]: Done

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:55]: Done

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:44]: Done

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:57]: Yeah, the default for GH is bash. Used that syntax.

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:66]: It is bash indeed

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:60]: but `:)` is way more fun 😄 
Alright done

File: .github/workflows/build.yml
Comment (mihaiplesa) [inline .github/workflows/build.yml:74]: Newline before. `GitHub`

**Review by mihaiplesa — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml
Comment (mihaiplesa) [inline .github/workflows/build.yml:10]: Would it work with `ubuntu-latest`?

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:74]: Done

**Review by mihaiplesa — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml
Comment (kim0) [reply .github/workflows/build.yml:75]: Done

**Pull request merged**
